src/runtime_src/core/common/api/xrt_device.cpp

```diff
@@ -380,6 +380,25 @@ operator== (const device& d1, const device& d2)
 ////////////////////////////////////////////////////////////////
 namespace xrt::aie {
 
+// User passes column index relative to the partition and context id
+// of that partition. Get patition info using the context id passed and
+// convert relative column index to absolute using this info
+static uint16_t
+get_abs_col(const xrt_core::device* device, uint16_t context_id, uint16_t col)
+{
+  auto data = xrt_core::device_query_default<xrt_core::query::aie_partition_info>(device, {});
+  for (const auto& entry : data) {
+    if (std::stoi(entry.metadata.id) != context_id)
+      continue;
+
+    auto abs_col = col + entry.start_col;
+    if (abs_col >= entry.num_cols)
+      throw std::out_of_range("col index out of range");
+    return abs_col;
```

Comment:
This is fine, but I meant to say that you can reuse `col` now that is passed by value.  The increment you did was fine just not as reference.

Reply:
Yes Soren understood what you meant but I thought this way reads well